src/datahelper.py

```diff
@@ -1,200 +1,404 @@
-                    #       Implement a if-check that only creates empty cols next to cols that contain strings such as "Europe" and not [1,2,3], which is used as bandwidth and there is nothing to annotate
-                    df.insert(
-                        loc=col_index + 1 + offset_index,
-                        column=col_header + "_isAbout",
-                        value="valueReference",
-                    )
-                    # offset index for column insertion to right for each added column, increase by 1 for each column
-                    offset_index += 1
-
-            self.to_csv(df_dict=(filename, df))
-
-    def create_json_dict_from_user_defined_columns(self):
-        # read columns and make dict with column names as keys and current timestamp as value
-
-        user_defined_cols_dict = {
-            (df_name): (set(value.columns.tolist()) - set(self.oedatamodel_col_list))
+        return {
+            (df_name): (set(value.columns.tolist()) - set(OEDATAMODEL_COL_LIST))
             for (df_name, value) in self.df_dict.items()
         }
 
-        # todo: Think about working with pd.DataFrame.from_dict here
+    def create_json_dict_from_user_defined_columns(self) -> dict:
+        """
+        Read columns and return dict with column names as keys and empty value.
+        :return: dict: Key -> filename; Value -> dict of user_defined_columns
+        """
+        user_defined_cols_dict = self.return_user_defined_columns()
+
         json_dict_user_col = {}
         for df_name, user_cols in user_defined_cols_dict.items():
-            csv_dict = {}
-            for user_col in user_cols:
-                csv_dict[user_col] = ""
-
+            csv_dict = {user_col: "" for user_col in user_cols}
             json_dict_user_col[df_name] = csv_dict
 
         return json_dict_user_col
 
-    def create_version_dict(self) -> dict:
+    def insert_user_column_dict_in_csv(self) -> None:
         """
-        The method inserts a version as value for each user-defined column. The version is the date of today.
-        :rtype: object
-        :return: version_dict:
+        Insert each csv-specific column dicts in respective csv.
+        :type columns: object
+        :param columns: Specify one of: version, other, all
+        :return:
         """
-
         json_dict_user_col = self.create_json_dict_from_user_defined_columns()
 
-        df = pd.DataFrame.from_dict(json_dict_user_col, orient="index")
+        for filename, df_data in self.df_dict.items():
+            for column in JSON_COL_LIST:
+                df_data[column] = f"{json_dict_user_col.get(filename)}"
 
-        df.where(df.isna(), (self.now.strftime("%d/%m/%Y")), inplace=True)
+            self.to_csv(df_dict=(filename, df_data))
 
-        version_dict = {k: v.dropna().to_dict() for k, v in df.T.items()}
+    def postgresql_conform_columns(self) -> dict:
+        """
+        Correct columns from csv files to be postgresql conform and save in csv.
```

Comment:
Instead of correcting headers, I would just check and raise error if there are non-conform headers.
Otherwise some magic  transforming happens in the background?
(And you don't have to check for similarity and so on...)